File: src/main/kotlin/org/example/service/NotebookService.kt
package org.example.service

import org.example.domain.Notebook
import org.example.domain.request.NotebookRequest
import java.util.*

interface NotebookService {
    fun getAll(): List<Notebook>

    fun getById(id: UUID): Notebook?

    fun save(notebook: NotebookRequest): Notebook

    fun delete(id: UUID)

    fun update(notebook: Notebook): Notebook
}

File: frontend/src/restClient/notebookService.js
import { AXIOS } from "../axios";

const NOTEBOOK_URL = '/notebook'

const authHeader = () => {
    return {
        headers: { Authorization: "Bearer " + localStorage.getItem('userToken') }
    }
}

// id title text
export default {
    getAll() {
        return AXIOS.get(`${NOTEBOOK_URL}/`, authHeader())
    },

    getById(id) {
        return AXIOS.get(`${NOTEBOOK_URL}/${id}`, authHeader())
    },

    save(notebook) {
        return AXIOS.post(`${NOTEBOOK_URL}/`, notebook, authHeader())
    },

    update(notebook) {
        return AXIOS.put(`${NOTEBOOK_URL}/`, notebook, authHeader())
    },

    delete(id) {
        return AXIOS.delete(`${NOTEBOOK_URL}/${id}`, authHeader())
    }
}